src/lib/cache/cacheManager.ts

```diff
@@ -0,0 +1,356 @@
+    await loadSinglePrefixCommandVersionToCache(newVersion);
+}
+
+export async function refreshAllPrefixCommandVersionsCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    // Step 1: Get all versions from the database
+    const prefixCommandVersions = await PrefixCommandVersion.find();
+    // Step 2: Get all versions from the cache
+    const cacheKeys = await inMemoryCache.store.keys();
+    // Step 3: Loop over cached versions
+    for (const key of cacheKeys) {
+        if (key.startsWith(`${memoryCachePrefixVersion}:`)) {
+            const checkVersion = key.split(':')[1];
+            // Step 3.a: Check if cached version exists in the database list
+            let found = false;
+            for (const dbVersion of prefixCommandVersions) {
+                const { _id: dbVersionId, alias } = dbVersion;
+                if (dbVersionId.toString().toLowerCase() === checkVersion.toLowerCase() || alias.toLowerCase() === checkVersion.toLowerCase()) {
+                    found = true;
+                    break;
+                }
+            }
+            // Step 3.b: If not found, remove from cache
+            if (!found) {
+                Logger.debug(`Removing version with id ${checkVersion} from cache`);
+                // eslint-disable-next-line no-await-in-loop
+                await inMemoryCache.del(key);
+            }
+        }
+    }
+    // Step 4: Loop over database versions and update cache
+    for (const dbVersion of prefixCommandVersions) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandVersionToCache(dbVersion);
+    }
+}
+
+/**
+ * Prefix Command Category Cache Management Functions
+ */
+
+export async function clearSinglePrefixCommandCategoryCache(category: IPrefixCommandCategory) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { name } = category;
+    Logger.debug(`Clearing cache for command category "${name}"`);
+    await inMemoryCache.del(`${memoryCachePrefixCategory}:${name.toLowerCase()}`);
+}
+
+export async function loadSinglePrefixCommandCategoryToCache(category: IPrefixCommandCategory) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { name } = category;
+    Logger.debug(`Loading category ${name} to cache`);
+    await inMemoryCache.set(`${memoryCachePrefixCategory}:${name.toLowerCase()}`, category.toObject());
+}
+
+export async function loadAllPrefixCommandCategoriesToCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    const PrefixCommandCategories = await PrefixCommandCategory.find();
+    for (const category of PrefixCommandCategories) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandCategoryToCache(category);
+    }
```

Comment:
```suggestion
    await Promise.all(PrefixCommandCategories.map((category) => loadSinglePrefixCommandCategoryToCache(category)));
```

Nerd level increasing... xD